File: src/main/java/com/blademc/core/Cosmetic/FX/util/RandomUtils.java
package com.blademc.core.Cosmetic.FX.util;

import cn.nukkit.block.Block;
import cn.nukkit.math.Vector3;

import java.util.Random;

public final class RandomUtils {

    public static final Random random = new Random(System.nanoTime());

    private RandomUtils() {
        // No instance allowed
    }

    public static Vector3 getRandomVector() {
        double x, y, z;
        x = random.nextDouble() * 2 - 1;
        y = random.nextDouble() * 2 - 1;
        z = random.nextDouble() * 2 - 1;

        return new Vector3(x, y, z).normalize();
    }

    public static Vector3 getRandomCircleVector() {
        double rnd, x, z;
        rnd = random.nextDouble() * 2 * Math.PI;
        x = Math.cos(rnd);
        z = Math.sin(rnd);

        return new Vector3(x, 0, z);
    }

    public static Block getRandomMaterial(Block[] materials) {
        return materials[random.nextInt(materials.length)];
    }

    public static double getRandomAngle() {
        return random.nextDouble() * 2 * Math.PI;
    }

}

File: src/main/kotlin/com/elissamc/particle/util/RandomUtils.kt
package com.elissamc.particle.util

import cn.nukkit.block.Block
import cn.nukkit.math.Vector3

import java.util.Random

object RandomUtils {

    val random = Random(System.nanoTime())

    val randomVector: Vector3
        get() {
            val x: Double = random.nextDouble() * 2 - 1
            val y: Double = random.nextDouble() * 2 - 1
            val z: Double = random.nextDouble() * 2 - 1

            return Vector3(x, y, z).normalize()
        }

    val randomCircleVector: Vector3
        get() {
            val rnd: Double = random.nextDouble() * 2.0 * Math.PI
            val x: Double
            val z: Double
            x = Math.cos(rnd)
            z = Math.sin(rnd)

            return Vector3(x, 0.0, z)
        }

    val randomAngle: Double
        get() = random.nextDouble() * 2.0 * Math.PI

    fun getRandomMaterial(materials: Array<Block>): Block {
        return materials[random.nextInt(materials.size)]
    }

}// No instance allowed